src/interactive-window/interactiveWindowProvider.ts

```diff
@@ -109,6 +102,37 @@ export class InteractiveWindowProvider
         asyncRegistry.push(this);
 
         this.notebookEditorProvider.registerEmbedNotebookProvider(this);
+        this.restoreWindows();
+    }
+
+    private restoreWindows() {
+        // VS Code controls if interactive windows are restored.
+        const interactiveWindowMapping = new Map<string, InteractiveTab>();
+        window.tabGroups.all.forEach((group) => {
+            group.tabs.forEach((tab) => {
+                if (isInteractiveInputTab(tab) && tab.input.uri) {
+                    interactiveWindowMapping.set(tab.input.uri.path, tab);
```

Comment:
I think we have something like getkey for a uri? Would that be better than path? (Path wouldn't match say if the cases were different on windows)

Reply:
@rchiodo I pushed a fix to use `uri.toString()`.

Reply:
Is that the same as getComparisonKey? https://github.com/microsoft/vscode-jupyter/blob/acb70fa84930ad77ccd08180b5351902664e04d4/src/platform/vscode-path/resources.ts#L57

Reply:
If we need to compare `Uri`s we should use `getComparisonKey` which can normalize paths, here we are using it as an identifier so a `toString()` is sufficient.